packages/notifier/src/storesub.js

```diff
@@ -2,9 +2,76 @@
 import { E } from '@endo/eventual-send';
 import { Far, makeMarshal } from '@endo/marshal';
 import { observeIteration } from './asyncIterableAdaptor.js';
+import { makePublishKit, subscribeEach } from './publish-kit.js';
 import { makeSubscriptionKit } from './subscriber.js';
 
 /**
+ * Begin iterating the source, storing serialized iteration values.  If the
+ * storageNode's `setValue` operation rejects, the iteration will be terminated.
+ *
+ * Returns a StoredSubscriber that can be used by a client to directly follow
+ * the iteration themselves, or obtain information to subscribe to the stored
+ * data out-of-band.
+ *
+ * @template T
+ * @param {Subscriber<T>} subscriber
+ * @param {ERef<StorageNode>} storageNode
+ * @param {ERef<ReturnType<typeof makeMarshal>>} marshaller
+ * @returns {StoredSubscriber<T>}
+ */
+export const makeStoredSubscriber = (subscriber, storageNode, marshaller) => {
+  assert(subscriber && storageNode && marshaller, 'missing argument');
+
+  // Abort the iteration on the next observation if the publisher ever fails.
+  let publishFailed = false;
+  let publishException;
+
+  const fail = err => {
+    publishFailed = true;
+    publishException = err;
+  };
+
+  // Must *not* be an async function, because it sometimes must throw to abort
+  // the iteration.
+  const publishValue = obj => {
+    if (publishFailed) {
+      // To properly abort the iteration, this must be a synchronous exception.
+      throw publishException;
```

Comment:
This still strikes me as a code smell, and I'm concerned about race conditions. It would seem much better to use a pattern that enforces serial consumption rather than the somewhat sloppy `observeIterator`.
```js
const iteratorP = E(subscribeEach(subscriber))[Symbol.asyncIterator]();
const consumeResultsSerially = async (iteratorP, consumeValue) => {
  let finished = false;
  while (!finished) {
    const { value, done } = await E(iteratorP).next();
    await consumeValue(value);
    finished = done;
  }
};
consumeResultsSerially(iteratorP, publishValue)
  // TODO: How should we handle and/or surface this failure?
  .catch(() => {});
```

Reply:
Good idea. PTAAL.